util/movement-types/src/block.rs

```diff
@@ -106,6 +106,23 @@ impl Block {
 	pub fn add_transaction(&mut self, transaction: Transaction) {
 		self.transactions.insert(transaction);
 	}
+
+	pub fn collapse(blocks: Vec<Block>) -> Block {
+		let mut transactions = BTreeSet::new();
+		let parent = if let Some(first_block) = blocks.first() {
+			first_block.parent
+		} else {
+			Id::genesis_block()
```

Comment:
Should we rather handle the empty case specially and return `Option<Block>` to avoid producing something out of nothing?

Reply:
All this function is as an example to explain the idea of batch aggregation.